components/instance-creator/internal/client/caller.go

```diff
@@ -0,0 +1,43 @@
+package client
+
+import (
+	"github.com/kyma-incubator/compass/components/director/pkg/auth"
+	"github.com/kyma-incubator/compass/components/director/pkg/securehttp"
+	"github.com/kyma-incubator/compass/components/instance-creator/internal/config"
+	"github.com/pkg/errors"
+)
+
+// CallerProvider is used to provide ExternalSvcCaller to call external services with given authentication
+type CallerProvider struct{}
+
+// NewCallerProvider creates new CallerProvider
+func NewCallerProvider() *CallerProvider {
+	return &CallerProvider{}
+}
+
+// GetCaller provides ExternalSvcCaller to call external services with given authentication
+func (c *CallerProvider) GetCaller(config config.Config, region string) (ExternalSvcCaller, error) {
```

Comment:
We should always return concrete structs and avoid returning interfaces to avoid need of refactoring in case more methods are added.

https://github.com/golang/go/wiki/CodeReviewComments#interfaces